ast/src/main/java/com/graphicsfuzz/common/ast/decl/ArrayInfo.java

```diff
@@ -17,37 +17,58 @@
 package com.graphicsfuzz.common.ast.decl;
 
 import com.graphicsfuzz.common.ast.IAstNode;
+import com.graphicsfuzz.common.ast.expr.Expr;
+import com.graphicsfuzz.common.ast.expr.IntConstantExpr;
 import com.graphicsfuzz.common.ast.visitors.IAstVisitor;
+import com.graphicsfuzz.common.ast.visitors.UnsupportedLanguageFeatureException;
 import java.util.Optional;
 
 public class ArrayInfo implements IAstNode {
 
-  private Optional<Integer> size;
+  private Optional<Expr> size;
+  private Optional<Expr> originalSize;
 
-  public ArrayInfo(Integer size) {
+  public ArrayInfo(Expr size) {
     this.size = Optional.of(size);
+    this.originalSize = Optional.of(size);
   }
 
   public ArrayInfo() {
     this.size = Optional.empty();
+    this.originalSize = Optional.empty();
   }
 
   public boolean hasSize() {
     return size.isPresent();
   }
 
-  public Integer getSize() {
+  public Integer getSize() throws UnsupportedLanguageFeatureException {
+    if (size.get() instanceof IntConstantExpr) {
+      return ((IntConstantExpr)size.get()).getNumericValue();
+    }
+    throw new UnsupportedLanguageFeatureException("Not a constant expression");
+  }
+
+  public Expr getSizeExpr() {
     return size.get();
   }
 
```

Comment:
Is there ever a use case for setting the size expr to something that is not an IntConstantExpr?  If not then I suggest calling this "setConstantSize" and making it take an IntConstantExpr, with a JavaDoc comment explaining the purpose of it.

Reply:
Changed parameter to int.